### PR TITLE
drivers: net: rswitch: Remove duplicate code

### PR DESCRIPTION
**Pull request by Deedone**
Move shared code from flower and matchall filters to common file.

Signed-off-by: Mykyta Poturai <mykyta_poturai@epam.com>

### CODE REVIEW

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c [hunk 1]
```diff
@@ -7,166 +7,78 @@
...
-		}
+	struct rswitch_tc_filter *filter = kzalloc(sizeof(*filter), GFP_KERNEL);
+	int rc;
 
-		if (entry->id == FLOW_ACTION_VLAN_MANGLE) {
-			if (be16_to_cpu(entry->vlan.proto) != ETH_P_8021Q) {
-				pr_err("Unsupported VLAN proto for offload!\n");
-				return -EOPNOTSUPP;
-			}
+	printk("t0\n");
```
Comment (LKomaryanskiy) [inline drivers/net/ethernet/renesas/rswitch_tc_matchall.c]: Remove debug code

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c [hunk 2]
```diff
@@ -7,166 +7,78 @@
...
 
-			filter.vlan_id = entry->vlan.vid;
-			filter.vlan_prio = entry->vlan.prio;
-			filter.action |= ACTION_VLAN_CHANGE;
-			continue;
-		}
+	filter->cookie = cls_matchall->cookie;
+	filter->rdev = rdev;
+	filter->param.priv = rdev->priv;
+	printk("t1\n");
```
Comment (LKomaryanskiy) [inline drivers/net/ethernet/renesas/rswitch_tc_matchall.c]: The same

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c [hunk 3]
```diff
@@ -7,166 +7,78 @@
...
 
-		return -EOPNOTSUPP;
-	}
+	rc = rswitch_tc_validate_action(rdev, rule);
+	if (rc)
+		goto free;
+	printk("t2\n");
```
Comment (LKomaryanskiy) [inline drivers/net/ethernet/renesas/rswitch_tc_matchall.c]: The same

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c [hunk 4]
```diff
@@ -7,166 +7,78 @@
...
-	if (filter.action & (ACTION_CHANGE_DMAC | ACTION_VLAN_CHANGE)) {
-		if (!(filter.action & ACTION_MIRRED_REDIRECT)) {
-			pr_err("DMAC and VLAN change is supported only with redirect\n");
-			return -EOPNOTSUPP;
-		}
-	}
+	rc = rswitch_tc_setup_action(filter, rule);
+	if (rc)
+		goto free;
+	printk("t3\n");
```
Comment (LKomaryanskiy) [inline drivers/net/ethernet/renesas/rswitch_tc_matchall.c]: The same

File: drivers/net/ethernet/renesas/rswitch_tc_common.c [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+	if (f->action & ACTION_DROP) {
+		return rswitch_tc_setup_drop_action(f);
+	}
+
+	if (f->action & ACTION_MIRRED_REDIRECT) {
+		return rswitch_tc_setup_redirect_action(f);
+	}
+
+	return -EOPNOTSUPP;
+}
```
Comment (LKomaryanskiy) [inline drivers/net/ethernet/renesas/rswitch_tc_common.c]: Add newline at the end of file

File: drivers/net/ethernet/renesas/rswitch_tc_common.h [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+#include "rswitch_tc_filters.h"
+
+int rswitch_tc_validate_action(struct rswitch_device *rdev,
+				struct flow_rule *rule);
+
+int rswitch_tc_setup_action(struct rswitch_tc_filter *f,
+				struct flow_rule *rule);
+
+
+#endif
```
Comment (LKomaryanskiy) [inline drivers/net/ethernet/renesas/rswitch_tc_common.h]: Add newline at the end of file

File: drivers/net/ethernet/renesas/rswitch_tc_matchall.c [hunk 7]
```diff
@@ -7,166 +7,78 @@
...
 
-	return rswitch_add_mall_action(&filter, cls_matchall);
+	rc = rswitch_add_mall_action(filter, cls_matchall);
+	if (rc)
+		goto free;
+	printk("t4\n");
```
Comment (LKomaryanskiy) [inline drivers/net/ethernet/renesas/rswitch_tc_matchall.c]: The same

File: drivers/net/ethernet/renesas/rswitch_tc_common.c [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+ *
+ * Copyright (C) 2022 Renesas Electronics Corporation
+ * Copyright (C) 2022 EPAM Systems
+ */
+
+#include "rswitch.h"
+#include "rswitch_tc_filters.h"
+#include "rswitch_tc_common.h"
+
+int rswitch_tc_validate_action(struct rswitch_device *rdev,
```
Comment (firscity) [inline drivers/net/ethernet/renesas/rswitch_tc_common.c]: Better to clarify and use something like `rswitch_tc_validate_flow_action`

**Review by LKomaryanskiy — CHANGES_REQUESTED** [review]

File: drivers/net/ethernet/renesas/rswitch_tc_common.c [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+			f->action = ACTION_DROP;
+			break;
+		case FLOW_ACTION_REDIRECT:
+			f->action |= ACTION_MIRRED_REDIRECT;
+			f->target_rdev = netdev_priv(act->dev);
+			break;
+		case FLOW_ACTION_MANGLE:
+			/*
+			 * The only FLOW_ACT_MANGLE_HDR_TYPE_ETH is supported,
+			 * sanitized by rswitch_tc_flower_validate_action().
```
Comment (firscity) [inline drivers/net/ethernet/renesas/rswitch_tc_common.c]: Update comment

File: drivers/net/ethernet/renesas/rswitch_tc_common.c [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+			break;
+		case FLOW_ACTION_VLAN_MANGLE:
+			f->action |= ACTION_VLAN_CHANGE;
+			f->vlan_id = act->vlan.vid;
+			f->vlan_prio = act->vlan.prio;
+			break;
+		default:
+			/*
+			 * Should not come here, such action will be dropped by
+			 * rswitch_tc_flower_validate_action().
```
Comment (firscity) [inline drivers/net/ethernet/renesas/rswitch_tc_common.c]: Update comment

File: drivers/net/ethernet/renesas/rswitch_tc_common.h [hunk 11]
```diff
@@ -0,0 +1,20 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Renesas Ethernet Switch Driver tc flower functions
```
Comment (firscity) [inline drivers/net/ethernet/renesas/rswitch_tc_common.h]: Update description

File: drivers/net/ethernet/renesas/rswitch_tc_common.c [hunk 12]
```diff
@@ -0,0 +1,155 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Renesas Ethernet Switch Driver tc matchall functions
```
Comment (firscity) [inline drivers/net/ethernet/renesas/rswitch_tc_common.c]: Copy-paste

File: drivers/net/ethernet/renesas/rswitch_tc_common.c [hunk 13]
```diff
@@ -0,0 +1,155 @@
...
+			f->param.l23_info.vlan_id = f->vlan_id;
+			f->param.l23_info.vlan_prio = f->vlan_prio;
+		}
+	}
+
+	return 0;
+}
+static int rswitch_tc_setup_drop_action(struct rswitch_tc_filter *f)
+{
+	f->param.slv = 0x3F;
```
Comment (firscity) [inline drivers/net/ethernet/renesas/rswitch_tc_common.c:99]: Probably this is a bug (should be only corresponding port, not all), but it is a point for another issue.

**Review by firscity — CHANGES_REQUESTED** [review]

**Review by firscity — APPROVED** [review]
Nice job

**Review by LKomaryanskiy — APPROVED** [review]
Good job. Approved.

**Pull request merged**